.github/workflows/autoclose.yml

```diff
@@ -14,38 +14,45 @@ jobs:
         with:
           github-token: ${{secrets.GITHUB_TOKEN}}
           script: |
-            const patches = [
-              "@@ -63,7 +63,6 @@ $RECYCLE.BIN/\n # Icon must end with two \\r\n Icon\n \n-\n # Thumbnails\n ._*\n ",
-              "@@ -63,7 +63,6 @@ $RECYCLE.BIN/\n # Icon must end with two \\r\n Icon\n \n-\n # Thumbnails\n ._*\n \n@@ -165,7 +164,6 @@ config/curriculum.json\n config/i18n/all-langs.js\n config/certification-settings.js\n \n-\n ### vim ###\n # Swap\n [._]*.s[a-v][a-z]",
-              "@@ -165,7 +165,6 @@ config/curriculum.json\n config/i18n/all-langs.js\n config/certification-settings.js\n \n-\n ### vim ###\n # Swap\n [._]*.s[a-v][a-z]"
-            ];
-            const files = await github.rest.pulls.listFiles({
+            if (
+              files.data.length !== 1 ||
+              files.data[0].filename !== ".gitignore"
+            ) {
+              return;
+            }
+            const creator = context.payload.sender.login;
+            const opts = github.rest.issues.listForRepo.endpoint.merge({
+              ...context.issue,
+              creator,
+              state: 'all'
+            });
+            const issues = await github.paginate(opts);
+
+            for (const issue of issues) {
+              if (issue.number === context.issue.number) {
+                continue;
+              }
+
+              if (issue.pull_request) {
+                return // Creator is already a contributor.
+              }
+            }
```

Comment:
The logic here is "if have PR and not this one, then must have contributed"?  Could we add a comment to that effect?  I spent too long trying to understand it.